### PR TITLE
feat: add power ups for mario 

### PR DESCRIPTION
**Pull request by isaias-rojas**
## Description

Add the state change when the player collides with a Power Up (this PR don't include animations or sprites)

## Changes Made

- [ ] Detect collisions between player and power ups and dispatch the specific event
- [ ] Add Mario State Big when Colliding with mushroom
- [ ] When Mario is Big and collide with enemy, make it invincible for 3 seconds
- [ ] When Mario Collides with star is invincible for 10 seconds
- [ ] When Mario Collides with Fire Flower it just stays big

## Checklist

- [ ] I have reviewed my own code changes.
- [ ] I have tested these changes locally.
- [ ] My code follows the project's coding style guidelines.
- [ ] I have updated the documentation accordingly.
- [ ] I have added appropriate labels to this pull request.

### CODE REVIEW

**Comment by JuanPabloArequipa** [comment]
All implementation is reducing excessively the game performance, it is not possible to play due to frozen screen.
@CatCoder36 Let me know if you experience the same. I leave some videos:

https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/135129192/02451125-42aa-42f0-ba74-d950847bf0ab


https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/135129192/93f3dcbe-4b44-407b-b91f-1a756ee3afb2


https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/135129192/aa93fe48-caa2-4154-87ee-9939f8ca4852

File: MarioGame/Source/Components/PlayerComponent.cs [hunk 1]
```diff
@@ -11,6 +11,12 @@ public class PlayerComponent : BaseComponent
         public bool DeathAnimationComplete { get; set; }
         public bool ShouldProcessDeath { get; set; }
         public bool IsTimeOver { get; set; }
+        public bool IsInvincibleAfterHit { get; set; }
```
Comment (CatCoder36) [inline MarioGame/Source/Components/PlayerComponent.cs:14]: Podria reemplazarse todos esos valores booleanos por un enum para simplificar el codigo

File: MarioGame/Source/Services/TimerService.cs [hunk 2]
```diff
@@ -1,19 +1,89 @@
...
+
+        public void Dispose()
         {
-            using Timer timer = new Timer(delay.TotalMilliseconds) { AutoReset = false };
-            timer.Elapsed += (sender, e) =>
+            if (!_disposed)
+            {
+                _timers.Clear();
+                _disposed = true;
+                Console.WriteLine("TimerService disposed." + _disposed);
```
Comment (CatCoder36) [inline MarioGame/Source/Services/TimerService.cs:71]: Borra los logs

File: MarioGame/Source/Systems/BlockSystem.cs [hunk 3]
```diff
@@ -45,6 +47,14 @@ public override void Update(GameTime gameTime, IEnumerable<Entity> entities)
         UpdateBlocks(questionBlockEntities, gameTime, mushroomEntities, startEntities, playerEntities, flowerEntities, coinEntities);
         UpdateBlocks(coinBlockEntities, gameTime, mushroomEntities, startEntities, playerEntities, flowerEntities, coinEntities);
 
+        if (_bodiesToRemove.Count == 0)
+        {
+            return;
+        }
+        Console.WriteLine("Destroying bodies " + _bodiesToRemove.Count);
```
Comment (CatCoder36) [inline MarioGame/Source/Systems/BlockSystem.cs:54]: Borra los logs

**Comment by CatCoder36** [comment]
Y sigue los parametros de nombre de rama, la rama debe contener tipo/id/nombre

**Comment by CatCoder36** [comment]
> All implementation is reducing excessively the game performance, it is not possible to play due to frozen screen. @CatCoder36 Let me know if you experience the same. I leave some videos:
> 
> After eating a Mushroom and collides with a Gomba :
> 
>  2024-06-28.22-30-34.mov 
> After eating a Star, kill some Gombas and jumping repeatedly:
> 
>  2024-06-28.22-35-07.mov 
> When there are some multilevel blocks and try to jump between them:
> 
>  2024-06-28.22-41-03.mov

Si, tengo los mismos problemas de rendimiento

**Pull request closed without merging**